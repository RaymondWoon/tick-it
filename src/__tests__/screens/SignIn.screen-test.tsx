/* src/__tests__/screens/Login.screen-test.tsx

/* Core */
import {
  act,
  fireEvent,
  render,
  screen,
  userEvent,
  waitFor,
} from "@testing-library/react-native";
//import { renderRouter } from "expo-router/testing-library";
// import {
//   act,
//   renderRouter,
//   screen as expo_screen,
//   waitFor,
// } from "expo-router/testing-library";

/* Components */
import SignIn from "#app/sign-in";

describe("<SignInScreen />", () => {
  /* Test the logo */
  it("Should render the logo", () => {
    render(<SignIn />);

    const logo = screen.getByTestId("logo");
    expect(logo).toBeTruthy();
  });

  /* Test the application title */
  it("Should render the title of the app", () => {
    render(<SignIn />);

    const title = screen.getByText("Welcome to Tick-it");
    expect(title).toBeTruthy();
  });

  /* Test the email input */
  it("Should render the email input", () => {
    render(<SignIn />);

    const emailInput = screen.getByPlaceholderText("Email");
    expect(emailInput).toBeTruthy();
  });

  /* Test the password input */
  it("Should render the password input", () => {
    render(<SignIn />);

    const passwordInput = screen.getByPlaceholderText("Password");
    expect(passwordInput).toBeTruthy();
  });

  /* Test the sign-in button */
  it("Should render the the sign-in button", () => {
    render(<SignIn />);

    const signinBtn = screen.getByText("Sign In");
    expect(signinBtn).toBeTruthy();
  });

  /* Test the forget password button */
  // it("Should render the forget password button", () => {
  //   render(<SignIn />);

  //   const forgetPwdBtn = screen.getByText("Forgot password?");
  //   expect(forgetPwdBtn).toBeTruthy();
  // });

  /* Test the sign up text */
  it("Should render the sign up button", () => {
    render(<SignIn />);

    const signupBtn = screen.getByText("Don't have an account? ");
    expect(signupBtn).toBeTruthy();
  });

  /* Test the sign-up button */
  it("Should render the sign up button", () => {
    render(<SignIn />);

    const signupBtn = screen.getByText("Sign Up");
    expect(signupBtn).toBeTruthy();
  });

  /* Test the Sign-up press event */
  //it("Should display the register screen", async () => {
  //render(<Login />);

  //const user = userEvent.setup();

  // renderRouter({
  //   "/login": Login,
  //   "/register": Register,
  // });

  // try {
  //   await user.press(screen.getByRole("Pressable", { name: "Sign-up" }));
  // } catch (error) {
  //   console.log(error);
  // }

  //const signupBtn = screen.getByText("Sign-up");
  //expect(signupBtn).toBeTruthy();

  //await user.press(signupBtn);

  //fireEvent.press(signupBtn);

  //expect(screen.getByText("Create account")).toBeOnTheScreen();

  //expect(screen).toHavePathname("/register");
  //});

  /* Snapshot */
  it("Renders SignIn screen correctly", () => {
    const tree = render(<SignIn />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
