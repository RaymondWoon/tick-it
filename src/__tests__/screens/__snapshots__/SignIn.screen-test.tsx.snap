// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInScreen /> Renders SignIn screen correctly 1`] = `
<View
  style={
    [
      {
        "flex": 1,
        "marginTop": 54,
      },
      {
        "backgroundColor": "#FDFFFC",
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "gap": 28.5,
          "paddingHorizontal": 41.5,
          "paddingTop": 38,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": 191,
          },
          {
            "backgroundColor": "#FDFFFC",
          },
        ]
      }
    >
      <Image
        alt="logo"
        source={
          {
            "testUri": "../../../assets/images/tick-it-logo.png",
          }
        }
        style={
          {
            "backgroundColor": "transparent",
            "height": 191,
            "width": 191,
          }
        }
        testID="logo"
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#011627",
              "fontSize": 34.5,
              "fontWeight": "400",
            },
            undefined,
            {
              "fontSize": 66.5,
              "fontWeight": "bold",
              "lineHeight": 61,
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Welcome to Tick-it
      </Text>
      <Text
        style={
          [
            {
              "color": "#011627",
              "fontSize": 34.5,
              "fontWeight": "400",
            },
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 41.5,
              "fontWeight": "bold",
            },
            undefined,
            undefined,
          ]
        }
      >
        The EZ task manager app
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderCurve": "continuous",
            "borderRadius": 30.5,
            "borderWidth": 1,
            "flexDirection": "row",
            "gap": 21,
            "height": 84,
            "justifyContent": "center",
            "paddingHorizontal": 31.5,
          },
          {
            "borderColor": "#CCD2D7",
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={49.5}
        bbWidth={49.5}
        className="at-thin__svg-icon-phosphor"
        color="#AAB3BB"
        fill="currentColor"
        focusable={false}
        height={49.5}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 49,
              "width": 49,
            },
          ]
        }
        testID="phosphor-react-native-at-thin"
        tintColor="#AAB3BB"
        vbHeight={256}
        vbWidth={256}
        width={49.5}
      >
        <RNSVGGroup
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M128 28a100 100 0 0 0 0 200c20.76 0 42.52-6.23 58.21-16.66a4 4 0 1 0-4.43-6.67C167.35 214.27 147.24 220 128 220a92 92 0 1 1 92-92c0 31.32-15 36-24 36s-24-4.68-24-36V88a4 4 0 0 0-8 0v14.75a44 44 0 1 0 2.82 45.94C171.46 163.58 181.66 172 196 172c20 0 32-16.45 32-44A100.11 100.11 0 0 0 128 28m0 136a36 36 0 1 1 36-36 36 36 0 0 1-36 36"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <TextInput
        autoCapitalize="none"
        icon={
          <At
            color="#AAB3BB"
            size={49.5}
            weight="thin"
          />
        }
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#aaa"
        style={
          [
            {
              "color": "#011627",
              "flex": 1,
              "fontSize": 37.5,
            },
          ]
        }
        textContentType="emailAddress"
        value=""
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderCurve": "continuous",
                "borderRadius": 30.5,
                "borderWidth": 1,
                "flexDirection": "row",
                "gap": 21,
                "height": 84,
                "justifyContent": "center",
                "paddingHorizontal": 31.5,
              },
              {
                "borderColor": "#CCD2D7",
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={49.5}
            bbWidth={49.5}
            className="lock-simple-thin__svg-icon-phosphor"
            color="#AAB3BB"
            fill="currentColor"
            focusable={false}
            height={49.5}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                [
                  undefined,
                  undefined,
                  {},
                ],
                {
                  "flex": 0,
                  "height": 49,
                  "width": 49,
                },
              ]
            }
            testID="phosphor-react-native-lock-simple-thin"
            tintColor="#AAB3BB"
            vbHeight={256}
            vbWidth={256}
            width={49.5}
          >
            <RNSVGGroup
              fill={
                {
                  "type": 2,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M208 84h-36V56a44 44 0 0 0-88 0v28H48a12 12 0 0 0-12 12v112a12 12 0 0 0 12 12h160a12 12 0 0 0 12-12V96a12 12 0 0 0-12-12M92 56a36 36 0 0 1 72 0v28H92Zm120 152a4 4 0 0 1-4 4H48a4 4 0 0 1-4-4V96a4 4 0 0 1 4-4h160a4 4 0 0 1 4 4Z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <TextInput
            autoCapitalize="none"
            icon={
              <LockSimple
                color="#AAB3BB"
                size={49.5}
                weight="thin"
              />
            }
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#aaa"
            secureTextEntry={true}
            style={
              [
                {
                  "color": "#011627",
                  "flex": 1,
                  "fontSize": 37.5,
                },
              ]
            }
            textContentType="password"
            value=""
          />
        </View>
      </View>
      <View
        style={
          {
            "justifyContent": "center",
            "marginLeft": 21,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        {
          "display": "flex",
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderCurve": "continuous",
          "height": 91.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderCurve": "continuous",
              "height": 91.5,
              "justifyContent": "center",
            },
            false,
            {
              "backgroundColor": "#2EC4B6",
              "borderRadius": 30.5,
              "width": 976.5,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 38,
              },
              {
                "color": "#FCFCFD",
              },
            ]
          }
        >
          <Text>
            Sign In
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#011627",
              "fontSize": 34.5,
              "fontWeight": "400",
            },
            {
              "fontSize": 33.5,
              "lineHeight": 46,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Don't have an account? 
      </Text>
      <Text
        href="/sign-up"
        onPress={[Function]}
        role="link"
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "gray",
              "borderRadius": 9.5,
              "borderWidth": 1,
              "justifyContent": "center",
              "marginLeft": 10.5,
              "padding": 4,
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            undefined,
          ]
        }
        testID="sign-up"
      >
        <Text
          style={
            {
              "color": "#011627",
            }
          }
        >
          Sign Up
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
